internal/server/audit/log/log.go

```diff
@@ -0,0 +1,126 @@
+package log
+
+import (
+	"context"
+
+	"go.flipt.io/flipt/internal/config"
+	"go.flipt.io/flipt/internal/server/audit"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+const (
+	sinkType = "log"
+	auditKey = "AUDIT"
+)
+
+// Sink is the structure in charge of sending audit events to a specified file location.
+type Sink struct {
+	logger *zap.Logger
+}
+
+type logOptions struct {
+	path     string
+	encoding config.LogEncoding
+}
+
+type Option func(*logOptions)
+
+func WithPath(path string) Option {
+	return func(o *logOptions) {
+		o.path = path
+	}
+}
+
+func WithEncoding(encoding config.LogEncoding) Option {
+	return func(o *logOptions) {
+		o.encoding = encoding
+	}
+}
+
+// NewSink is the constructor for a Sink.
+func NewSink(opts ...Option) (audit.Sink, error) {
+	oo := &logOptions{}
+
+	for _, o := range opts {
+		o(oo)
+	}
+
+	return newSink(*oo)
+}
+
+var (
+	consoleEncoderConfig = zapcore.EncoderConfig{
+		TimeKey:        "T",
+		LevelKey:       zapcore.OmitKey,
+		NameKey:        "N",
+		CallerKey:      zapcore.OmitKey,
+		FunctionKey:    zapcore.OmitKey,
+		MessageKey:     "M",
+		StacktraceKey:  zapcore.OmitKey,
+		LineEnding:     zapcore.DefaultLineEnding,
+		EncodeLevel:    zapcore.CapitalColorLevelEncoder,
+		EncodeTime:     zapcore.RFC3339TimeEncoder,
+		EncodeDuration: zapcore.StringDurationEncoder,
+		EncodeCaller:   zapcore.ShortCallerEncoder,
+	}
+
+	fileEncoderConfig = zapcore.EncoderConfig{
+		TimeKey:        zapcore.OmitKey,
+		LevelKey:       zapcore.OmitKey,
+		NameKey:        zapcore.OmitKey,
+		CallerKey:      zapcore.OmitKey,
+		FunctionKey:    zapcore.OmitKey,
+		MessageKey:     zapcore.OmitKey,
+		StacktraceKey:  zapcore.OmitKey,
+		LineEnding:     zapcore.DefaultLineEnding,
+		EncodeLevel:    zapcore.CapitalColorLevelEncoder,
+		EncodeTime:     zapcore.RFC3339TimeEncoder,
+		EncodeDuration: zapcore.StringDurationEncoder,
+		EncodeCaller:   zapcore.ShortCallerEncoder,
+	}
+)
+
+// newSink is the constructor for a Sink visible for testing.
+func newSink(opts logOptions) (audit.Sink, error) {
+	cfg := zap.Config{
+		Level:            zap.NewAtomicLevelAt(zapcore.InfoLevel),
+		Development:      false,
+		Encoding:         "console",
+		EncoderConfig:    consoleEncoderConfig,
+		OutputPaths:      []string{"stdout"},
+		ErrorOutputPaths: []string{"stderr"},
+	}
+
+	if opts.encoding != "" {
+		cfg.Encoding = string(opts.encoding)
+	}
+
+	if opts.path != "" {
+		cfg.OutputPaths = []string{opts.path}
```

Comment:
just realized zap doesnt create paths/folders if they don't already exist. going to move this back to draft until I fix that